sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/MSIToken.java

```diff
@@ -90,12 +84,13 @@ private static Long parseToEpochSeconds(String expiresOn, String expiresIn) {
         // expiresOn = timestamp of refresh expressed as seconds since epoch.
 
         // if we have an expiresOn, we'll use it. Otherwise, we use expiresIn.
-        String dateToParse = CoreUtils.isNullOrEmpty(expiresOn) ? expiresIn : expiresOn;
+        boolean isExpiresOn = !CoreUtils.isNullOrEmpty(expiresOn);
+        String dateToParse = !isExpiresOn ? expiresIn : expiresOn;
```

Comment:
```suggestion
        String dateToParse = isExpiresOn ? expiresOn: expiresIn;
```

There's enough negativity in the world.